### PR TITLE
fix overflowing contract names and origins

### PR DESCRIPTION
**Pull request by brad-decker**
Moves heading and subtitle into divs with h3/h2 children so that the
div can be display flex and still have ellipses overflow. Only the
heading was display flex but I wanted the two to have similar structure.
this allows subheading to be display flex in the future.

Also uses stripHttpSchemes to remove that from origin in the subheading

<details>
<summary>Screenshots</summary>

Before:<br />

After: <br/>

</details>

### CODE REVIEW

**Comment by brad-decker** [comment]
[Fix overflow of action name and origin on the transaction list](https://trello.com/c/WwpScBVg/87-fix-overflow-of-action-name-and-origin-on-the-transaction-list)

**Comment by rachelcope** [comment]
This looks great to me 🚀

**Review by rekmarks — CHANGES_REQUESTED** [review]
This looks much better!

Following recent examples of URL overflow handling, inspired by Chrome's URL style guide, the text-direction of overflowing URLs should be `rtl`, so that the ellipsis appears on the left, and the TLD is visible.

Ref: https://github.com/MetaMask/metamask-extension/blob/develop/ui/app/components/app/connected-sites-list/index.scss#L26-L33

**Comment by brad-decker** [comment]
@rekmarks so that would result in the status label/date being truncated instead of the domain?

**Comment by Gudahtt** [comment]
I was just going to suggest the same thing!

It would require moving the `origin` to a separate element, so it could be truncated separately. I don't think it makes sense to truncate the date.

**Comment by brad-decker** [comment]
or, is this suggestion to simply replace the first x characters of the domain with '...' and keep the status/date intact

**Comment by Gudahtt** [comment]
As an alternative, you could truncate it entirely in JavaScript, but I don't see how that could be easily done without some fancy DOM measurement, so... probably easier to use a separate element 🤔

**Comment by brad-decker** [comment]
Going to take some re-thinking no matter how it's done. This change has the truncation applied at the list-item level. In order to do this, I am going to have to override those defaults from the transaction-list-item level because not all list-items will have domains in the subtitle...

This means I'll likely need to change the type of the subtitle prop from just being a standard string to being an element.. which might be okay because currently there is the subtitleStatus node prop which may not be used outside of the transaction-list-item.

The next complexity is that not every type of transaction-list-item has an origin in the subtitle so we'd then have to do some checking...

Would anyone be opposed to the work involved in this being moved to a separate PR? I can open an issue to track it. The behavior in this PR is the correct default for everywhere there isn't a domain in the subtitle. cc @rekmarks @Gudahtt

**Comment by Gudahtt** [comment]
I think it's fine to address separately. I don't like truncating the more-important part of the origin anywhere if we can help it, but in this case it's really not a security-sensitive part of the UI so... it's not the end of the world.

**Review by Gudahtt — DISMISSED** [review]

**Comment by brad-decker** [comment]
Meh i guess it wasn't as difficult as i expected... 

<details>
<summary>Screenshot</summary>

Before:


After:




</details>

I'm gonna clean this up. the code that produces the above screenshot is not yet a part of this PR but will be after cleanup. @Gudahtt @rekmarks

**Review by rekmarks — DISMISSED** [review]
LGTM!

File: ui/app/components/app/transaction-list-item/index.scss [hunk 1]
```diff
@@ -31,4 +31,26 @@
...
+  }
+
+  &__origin, &__address {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+  }
+
+  &__origin {
+    direction: rtl;
```
Comment (Gudahtt) [inline ui/app/components/app/transaction-list-item/index.scss:54]:
```suggestion
    /*rtl:ignore*/
    direction: rtl;
```
We want this to be right-to-left even on right-to-left locales, but the build script to generate the rtl CSS file will flip this unless we tell it not to. This directive will ensure it stays as RTL for all locales

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**